packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -98,7 +103,7 @@ export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
        * */
       &:checked + span {
         background: ${theme.colors.primary.main};
-        border: none;
+        border: 1px solid ${theme.colors.primary.main};
```

Comment:
Adding the border here and in indeterminate moves the check/indeterminate bar a little bit, we need to tweak their absolute position if we are adding this border. Also disabled now has a blue border when it shouldn't.
<img width="61" alt="Screenshot 2023-05-15 at 16 39 42" src="https://github.com/grafana/grafana/assets/100691367/5054a316-19cd-478d-95f3-e9778501eb17">